app/models/semester_period.rb

```diff
@@ -0,0 +1,42 @@
+require 'sirius/enums/semester_period_type'
+require 'parity'
+
+class SemesterPeriod < Sequel::Model
+
+  many_to_one :faculty_semester
+
+  def type
+    Sirius::SemesterPeriodType.from_numeric(super)
+  end
+
+  def type=(new_type)
+    super Sirius::SemesterPeriodType.to_numeric(new_type)
+  end
+
+  def first_week_parity
+    if super
+      Parity.from_numeric(super)
```

Comment:
This is a property getter which converts number to symbol. Since the enum module requires a value, it is wrapped in case the value is nil. This is covered by specs.